src/gt4py/cartesian/gtc/debug/debug_codegen.py

```diff
@@ -0,0 +1,354 @@
+    def __init__(self) -> None:
+        self.body = utils.text.TextBlock()
+        self.symbol_table: dict[str, FieldDecl] = {}
+
+    def visit_Stencil(self, stencil: Stencil, **_):
+        self.generate_imports()
+
+        self.generate_run_function(stencil)
+
+        field_extents, block_extents = self.compute_extents(stencil)
+        self.symbol_table = self.initial_declarations(stencil, field_extents)
+        self.generate_stencil_code(stencil, block_extents)
+
+        return self.body.text
+
+    def generate_imports(self):
+        self.body.append("import numpy as np")
+        self.body.append("from gt4py.cartesian.gtc import ufuncs")
+        self.body.append("from gt4py.cartesian.utils import Field")
+
+    @staticmethod
+    def compute_extents(node: Stencil, **_) -> tuple[dict[str, Extent], dict[int, Extent]]:
+        ctx: StencilExtentComputer.Context = StencilExtentComputer().visit(node)
+        return ctx.fields, ctx.blocks
+
+    def initial_declarations(
+        self, stencil: Stencil, field_extents: dict[str, Extent]
+    ) -> dict[str, FieldDecl]:
+        self.body.append("# ===== Domain Description ===== #")
+        self.body.append("i_0, j_0, k_0 = 0,0,0")
+        self.body.append("i_size, j_size, k_size = _domain_")
+        self.body.empty_line()
+        self.body.append("# ===== Temporary Declaration ===== #")
+        symbol_table = self.generate_temp_decls(stencil.declarations, field_extents)
+        self.body.empty_line()
+        self.body.append("# ===== Field Declaration ===== #")
+        symbol_table |= self.generate_field_decls(stencil.params)
+        self.body.empty_line()
+        return symbol_table
+
+    def generate_temp_decls(
+        self, temporary_declarations: list[Temporary], field_extents: dict[str, Extent]
+    ) -> dict[str, FieldDecl]:
+        symbol_table: dict[str, FieldDecl] = {}
+        for declaration in temporary_declarations:
+            self.body.append(self.visit(declaration, field_extents=field_extents))
+            symbol_table[str(declaration.name)] = declaration
+        return symbol_table
+
+    def generate_field_decls(self, declarations: list[Decl]) -> dict[str, FieldDecl]:
+        symbol_table = {}
+        for declaration in declarations:
+            if isinstance(declaration, FieldDecl):
+                self.body.append(
+                    f"{declaration.name} = Field({declaration.name}, _origin_['{declaration.name}'], "
+                    f"({', '.join([str(x) for x in declaration.dimensions])}))"
+                )
+                symbol_table[str(declaration.name)] = declaration
+        return symbol_table
+
+    def generate_run_function(self, stencil: Stencil):
+        function_signature = "def run(*"
+        args = []
+        for param in stencil.params:
+            args.append(self.visit(param))
+        function_signature = ",".join([function_signature, *args])
+        function_signature += ",_domain_, _origin_):"
```

Comment:
```suggestion
        function_signature += ",".join(args)
        function_signature += ",_domain_, _origin_):"
```
would this be more clear to read since we are relying on `+=` to add origin and domain?

Reply:
There is the one awkwardness of it generating `def run(*,_domain_, _origin_)` in a non arg world with this approach

Reply:
hmm okay - then I guess this is as good as it gets. all good.